### DISCUSSION TITLE
`MSB3823` issues in `net4.6.1` SDK-style project

### DISCUSSION THREAD

**Discussion by ero-qt** [dotnet/msbuild#9970] [Q&A] [answered]
I'm in the middle of migrating some legacy projects to SDK-style. I'm getting `MSB3822` and `MSB3823`, which I fixed by doing the below:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <!-- ... -->
    <GenerateResourceUsePreserializedResources>true</GenerateResourceUsePreserializedResources>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="System.Resources.Extensions" Version="4.6.0" />
  </ItemGroup>
  <!-- ... -->
</Project>
```

However, this results in a much, much larger output upon building. I'm talking over 100 `System.*.dll` files added to the output compared to what was there before using the legacy style project. Why? How can I prevent this?

**Comment by GangWang01** [comment]
@just-ero can you provide a sample project to help to reproduce? It’s better to provide the build binary log (see how to [Providing MSBuild Binary Logs for investigation](https://github.com/dotnet/msbuild/blob/main/documentation/wiki/Providing-Binary-Logs.md#providing-msbuild-binary-logs-for-investigation)). Be aware that the binary log may capture sensitive information.

**Reply by ero-qt** [reply, marked as answer]
Hello again,

I have just now realized that this is a different discussion than I anticipated. I did not check it properly beforehand.

This issue was likely the result of including a NuGet package that did not target the parent project's target framework, resulting in extra runtime assemblies being required for that specific framework version.
